nixos/modules/i18n/input-method/fcitx5.nix

```diff
@@ -30,11 +69,30 @@ in
   config = mkIf (im.enabled == "fcitx5") {
     i18n.inputMethod.package = fcitx5Package;
 
+    environment.etc =
+      let
+        optionalFile = p: f: v: lib.optionalAttrs (v != { }) {
+          "xdg/fcitx5/${p}".text = f v;
```

Comment:
Can the system wide XDG_CONFIG_HOME be changed by environment variable or other method? Is there a better way to set it? Or can you read fcitx5 to check how it determine this file path such as reading from hard coded path or from others.

Reply:
i'm not sure what you mean. Do you want to ask why this works? The related code is https://github.com/fcitx/fcitx5/blob/master/src/lib/fcitx-utils/standardpath.cpp#L81.

Reply:
Allright, `/etc/xdg` is hard-coded in the source instead of some variables.